### PR TITLE
Matrix Printing: Fix printing errors in Matrix Symbol and subs()

### PR DESCRIPTION
**Pull request by prateeksingh0001**
Fixed the bug in _print_MatrixElement method so as to correctly
output the results of operations between two matrices.

Fixes https://github.com/sympy/sympy/issues/11821

### CODE REVIEW

**Comment by certik** [comment]
There seem to be some test errors, e.g. https://travis-ci.org/sympy/sympy/jobs/209281069

File: sympy/printing/tests/test_fcode.py [hunk 1]
```diff
@@ -675,3 +675,29 @@ def test_fcode_For():
...
+
+    E = A - B + M
+    F = C[0, 0]
+    F = F.subs(C, E)
+    assert (fcode(F) == "      ((-1)*B + A + M)(1, 1)")
+
+    E = A + M
+    F = C[0, 1]
+    F = F.subs(C, E)
+    assert (fcode(F) == "      (A + M)(1, 2)")
```
Comment (certik) [inline sympy/printing/tests/test_fcode.py]: I think this syntax is not allowed in Fortran:
```fortran
integer, parameter :: dp = kind(0d0)
real(dp) :: A(2,2), M(2,2)
A = 1
M = 2
print *, (A+M)(1, 2)
end
```
and compile using:
```
$ gfortran a.f90 
a.f90:6:14:

 print *, (A+M)(1, 2)
              1
Error: Syntax error in PRINT statement at (1)
```
```

**Comment by certik** [comment]
Probably better would be to distribute the [0,0] to each matrix, and not use (), per my comment about the `fcode` above.

**Comment by gxyd** [comment]
Closing this PR, since the issue has been fixed by #12800.

**Pull request closed without merging**